# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 500  # in Hz

# Default Controllers ------------------------------------------------
TorqueController:
  type: effort_controllers/JointGroupEffortController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7
  params:
    end_effector: link_ee

PositionTorqueController:
  type: effort_controllers/JointGroupPositionController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7
  joint_1:
    pid: {p: 800.0, i: 100, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_2:
    pid: {p: 800.0, i: 1000, d: 100.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_3:
    pid: {p: 800.0, i: 10, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_4:
    pid: {p: 800.0, i: 10, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_5:
    pid: {p: 300.0, i: 5, d: 10.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_6:
    pid: {p: 50.0, i: 1, d: 1.0, i_clamp_min: -10000, i_clamp_max: 10000}
  joint_7:
    pid: {p: 10.0, i: 0.1, d: 1.0, i_clamp_min: -10000, i_clamp_max: 10000}

PositionController:
  type: position_controllers/JointGroupPositionController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7
  pid_gains:
    joint_1: {p: 800.0, i: 100, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_2: {p: 800.0, i: 1000, d: 100.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_3: {p: 800.0, i: 10, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_4: {p: 800.0, i: 10, d: 80.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_5: {p: 300.0, i: 5, d: 10.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_6: {p: 50.0, i: 1, d: 1.0, i_clamp_min: -10000, i_clamp_max: 10000}
    joint_7: {p: 10.0, i: 0.1, d: 1.0, i_clamp_min: -10000, i_clamp_max: 10000}

# Custom Controllers ------------------------------------------------
# The space variable can be either set to "task" or "joint"
CustomControllers:
  type: iiwa_control/CustomEffortController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7
  params:
    space: task
    # end_effector is only needed in task space
    end_effector: link_ee
    null_space:
      joints: [0.044752691045324394, 0.6951627023357917, -0.01416978801753847, -1.0922311725109015, -0.0050429618456282, 1.1717338014778385, -0.01502630060305613]
      Kp: 20.
      Kd: 0.1
      max_torque: 10.
  controllers:
    # LinearDS: {type: "LinearDSController", params: [1., 1., 1.]}
    # ForceDS: {type: "ForceModDSController", params: [200]}
    PassiveDS: {type: "PassiveDSController", params: [180, 180]}
    PD: {type: "PidController", params: [20., 8.9], input: "Orientation", output: "Torque"}
  # structure:
    # AddCtrl:
    #   - LinearDS
    #   - ForceDS
    # CascadeCtrl:
    #  - LinearDS
    #  - PassiveDS

PositionTrajectoryController:
  type: position_controllers/JointTrajectoryController
  joints: 
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7

  constraints:
    goal_time: 0.5                   # Override default

  state_publish_rate:  25            # Override default
  action_monitor_rate: 30            # Override default
  stop_trajectory_duration: 0        # Override default

